pkg/aws/common/common.go

```diff
@@ -0,0 +1,76 @@
+package common
+
+import (
+	"errors"
+	"regexp"
+	"strings"
+
+	"github.com/aws/smithy-go"
+
+	awscostexplorer "github.com/aws/aws-sdk-go-v2/service/costexplorer"
+	awsclient "github.com/grafana/cloudcost-exporter/pkg/aws/client"
+)
+
+// Helpers
+// Parse the billing data into shared billing structure
+func ParseBilling(outputs []*awscostexplorer.GetCostAndUsageOutput, componentReStr string) *awsclient.BillingData {
+	componentRe := regexp.MustCompile(componentReStr)
+	b := &awsclient.BillingData{Regions: make(map[string]*awsclient.PricingModel)}
+	for _, out := range outputs {
+		for _, r := range out.ResultsByTime {
+			for _, g := range r.Groups {
+				if len(g.Keys) < 2 {
+					continue
+				}
+				service := g.Keys[0]
+				// Omit any tax lines entirely
+				if strings.EqualFold(service, "Tax") || strings.Contains(strings.ToLower(service), "tax") {
+					continue
+				}
+				usageKey := g.Keys[1]
+				region := getRegionFromKey(usageKey)
+				component := getComponentFromKey(usageKey)
+				// Only include NAT Gateway usage lines
+				if !componentRe.MatchString(component) {
+					continue
+				}
+				// Prefix component with service to carry it forward
+				b.AddMetricGroup(region, service+"|"+component, g)
+			}
+		}
+	}
+	return b
+}
+
+func getRegionFromKey(key string) string {
+	split := strings.Split(key, "-")
+	if len(split) < 2 {
+		return ""
+	}
+	billingRegion := split[0]
+	if region, ok := awsclient.BillingToRegionMap[billingRegion]; ok {
+		return region
+	}
+	return ""
+}
+
+func getComponentFromKey(key string) string {
+	split := strings.Split(key, "-")
+	if len(split) < 2 {
+		return ""
+	}
+	return strings.Join(split[1:], "-")
+}
+
+// isBillExpirationError determines whether the error returned by Cost Explorer
+// indicates that the bill changed during pagination and the request should be retried.
+func IsBillExpirationError(err error) bool {
+	var apiErr smithy.APIError
+	if errors.As(err, &apiErr) {
+		if apiErr.ErrorCode() == "BillExpirationException" {
```

Comment:
Try this:
```suggestion
		if apiErr.ErrorCode() == awscostexplorer.ErrCodeBillExpirationException {
```

Reply:
there seems to be a diff between these two ->
```
      awscostexplorer "github.com/aws/aws-sdk-go-v2/service/costexplorer"
+     "github.com/aws/aws-sdk-go/service/costexplorer"
```
the latter does have that symbol, but
```
go: module github.com/aws/aws-sdk-go is deprecated: aws-sdk-go is deprecated. Use aws-sdk-go-v2.
```

Reply:
but then, even after `go get ` blah/etc forcing _also_ this older sdk, ends up being the same string ->
```
    // The requested report expired. Update the date interval and try again.
    ErrCodeBillExpirationException = "BillExpirationException"
```

Reply:
Ah they had it in SDK v1 https://github.com/aws/aws-sdk-go/blob/v1.55.7/service/costexplorer/errors.go#L22 😞